packages/clerk-js/src/core/clerk.ts

```diff
@@ -1054,15 +1054,35 @@ export class Clerk implements ClerkInterface {
     await onAfterSetActive();
   };
 
-  #handlePendingSession = async (session: PendingSessionResource) => {
+  #handlePendingSession = async (
+    session: PendingSessionResource,
+    organization?: string | OrganizationResource | null | undefined,
+  ) => {
     if (!this.environment) {
       return;
     }
 
+    const shouldSwitchOrganization = organization !== undefined;
+    if (shouldSwitchOrganization) {
```

Comment:
Potentially thinking of extracting this bit of organization resolution logic to a separate private method in order to not duplicate on `setActive`